cryptostore/data/gd.py

```diff
@@ -4,148 +4,157 @@
+                                   'parents': [p_folder_id]}
+                batch.add(files.create(body=folder_metadata, fields='id, name'))
+            batch.execute()
+        self.folders = existing_as_dict
+
+
+    def write(self, bucket: str, path: str, file_name: str, **kwargs):
+        """
+        Upload file to Google Drive. File is stored in a parent folder which
+        name is that of 'path', replacing '/' with '-'. Folder name is
+        generated in `__init__`.
+
+        Parameters:
+            bucket:
+                Unused parameter.
+            path (str):
+                path = '{exchange}/{data_type}/{pair}/
                         {exchange}-{data_type}-{pair}-{int(timestamp)}.parquet'
-            String from which is derived folder name into which is written the
-            file, as well as the root folder that will contain all these
-            folders.
-        file_name (str):
-            File name preceded by path on local disk.
-        creds (str):
-            Path to credential file.
-
-    """
-    gah = StorageEngines['import googleapiclient.http']
-
-    # Retrieve folder ID to be used to write the file into, and upload.
-    # If not existing, folder will be created.
-    drive = _get_drive(creds)
-    folder_id, folder_name = _get_folder_in_parent(drive, path)
-    media = gah.MediaFileUpload(file_name, resumable=True)
-    file_metadata = {'name': path.split('/')[-1], 'parents': [folder_id]}
-    request = drive.files().create(body=file_metadata, media_body=media,
-                                   fields='id')
-    response = None
-    while response is None:
-        status, response = request.next_chunk(num_retries=4)
-
-    return
+                String from which is derived folder name into which is written
+                the file.
+            file_name (str):
+                File name preceded by path on local disk.
+
+        """
+
+        httplib2 = StorageEngines['httplib2']
+
+        # Retrieve folder ID to be used to write the file into.
+        # Get folder name first.
+        folder_name = '-'.join(path.split('/')[0:3])
+        folder_id = self.folders[folder_name]
+        # Upload (caching the authorized `http` object used to run
+        # `next_chunk()`)
+        googleapiclient = StorageEngines['googleapiclient.http']
+        media = googleapiclient.http.MediaFileUpload(file_name, resumable=True)
+        file_metadata = {'name': path.split('/')[-1], 'parents': [folder_id]}
+        request = self.drive.files().create(body=file_metadata,
+                                            media_body=media, fields='id')
+        googleapiclient = StorageEngines['googleapiclient._auth']
+        auth_http = googleapiclient._auth.authorized_http(self.creds)
+        auth_http.cache = httplib2.FileCache(self.cache_path)
+        response = None
+        while response is None:
+            status, response = request.next_chunk(num_retries=4, http=auth_http)
+
+        return
```

Comment:
this is not needed

Reply:
Hi Bryant,
these lines are needed in case file is bigger than default chunk size: file is then sent in chunk. This is the meaning of `resumable=True` in
```python
media = googleapiclient.http.MediaFileUpload(file_name, resumable=True)
```
I would prefer making no assumption regarding the file size the user will upload and keep these lines.
Please, why are they troubling you?

Reply:
the comment was on the `return` statement. A single `return` with no value/variable at the end of a function is superfluous

Reply:
Hi Bryant,
Ok. I just removed it.